### PR TITLE
Feature/containerise global

### PR DESCRIPTION
**Pull request by nicholac**

### CODE REVIEW

File: containers/web-server/Dockerfile-dev [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+RUN npm install
+# copy folders and files required for build into image N.B. when docker copy is given a
+# directory, it copies the contents but not the dir itself, so you must name the destination dir
+#   COPY images images
+#   COPY public public
+#   COPY src src
+#   COPY .eslintrc.js tsconfig.json tsconfig-typings.json .
+COPY frontend .
+# build to static site
+RUN npm run build
```
Comment (tomalrussell) [inline containers/web-server/Dockerfile-dev:23]: Perhaps this is comment to turn into an issue and put lower down the dev-experience TODO-list.. could we mount the source directory and run `npm start` to pick up frontend changes and live reload? Would need a fatter single-stage container, possibly just with node and react-scripts configured to proxy connections to other services.

File: deploy/README.md [hunk 2]
```diff
@@ -1,22 +1,15 @@
...
-- Vector tileserver, tileserver-gl-light, depends on nodejs
-- Raster tileserver, terracotta, depends on gunicorn and Python
-
-The frontend application source code is held in
-[this repository](https://github.com/nismod/infra-risk-vis/) and this guide
-assumes that it is built using node and npm locally on a development machine.
-It would be possible to build directly on the server in a working directory.
+The stack consists of five services:
+- Web server (nginx)
+- Raster tileserver (terracotta)
```
Comment (tomalrussell) [inline deploy/README.md:7]: Now folded into the `backend` app?

File: deploy/README.md [hunk 3]
```diff
@@ -45,122 +38,20 @@ terraform apply # rerun after any change to main.tf
...
-```
-
-Copy secret `PG*` variables for connection to database in EnvironmentFile as
-configured in backend.service, something like:
-
-```conf
 PGDATABASE=jamaica
 PGUSER=docker
 PGPASSWORD=docker
 PGHOST=localhost
```
Comment (tomalrussell) [inline deploy/README.md:51]: Note MySQL details here too? - or point to the `backend/README.md` which has more info.

File: deploy/main.tf [hunk 4]
```diff
@@ -3,37 +3,58 @@
...
+  name       = "global"
+  subnet_ids = module.vpc.public_subnets
+
   tags = {
-    Name = "Jamaica VPC"
+    Name = "Global VPC"
   }
 }
 
 resource "aws_security_group" "access_http_ssh" {
```
Comment (tomalrussell) [inline deploy/main.tf:55]: Do we need both this security group and the one defined from line 85?

File: deploy/main.tf [hunk 5]
```diff
@@ -3,37 +3,58 @@
...
-  name = "access_jamaica"
-  vpc_id = aws_default_vpc.jamaica.id
+  name   = "access_global"
+  vpc_id = "vpc-b39bfcdb"
```
Comment (tomalrussell) [inline deploy/main.tf:57]: Should this refer to `module.vpc.vpc_id`?

**Review by tomalrussell — APPROVED** [review]
Thanks for this PR - really helpful chunk of work to set up containerised dev and deployment, great progress on additional hazard layers.

I've tried to pick up small snags in the comments, nothing blocking.

File: containers/web-server/Dockerfile-dev (hunk 1 above)
Comment (nicholac) [reply containers/web-server/Dockerfile-dev:23]: I added live reload into the frontend last week and its been working well for me here using npm start outside the dev container (accessing backend services hosted within docker and exposed to localhost)

File: deploy/main.tf (hunk 4 above)
Comment (nicholac) [reply deploy/main.tf:55]: So this was a left-over from the first deployment which failed to re-deploy due to a known bug with TF relating to removal / alteration of security group names:  https://github.com/hashicorp/terraform-provider-aws/issues/265

I had to include both in order to enable the redeployment.

Realistically this whole file should probably not be in public source control - a seperate point but both should probably be tidied up.

Comment (tomalrussell) [reply deploy/main.tf:55]: Okay - for this and the below, let's check in on Monday, likely leave on the pile for now.

**Pull request merged**